src/uct/ib/base/ib_iface.c

```diff
@@ -764,16 +796,26 @@ static int uct_ib_iface_dev_addr_is_reachable(uct_ib_iface_t *iface,
         }
 
         /* Check FLID route: is enabled locally, and remote GID has it */
-        return (uct_ib_iface_is_flid_enabled(iface) &&
-                uct_ib_iface_gid_extract_flid(&params.gid) != 0);
+        if (!uct_ib_iface_is_flid_enabled(iface) ||
+            !uct_ib_iface_gid_extract_flid(&params.gid)) {
+            uct_iface_populate_info_str_buf(is_reachable_params,
+                                            "invalid flid route");
```

Comment:
`FLID`

Reply:
1. pls use uct_ib_iface_gid_extract_flid == 0 (instead of !uct_ib_iface_gid_extract_flid)
2. "IB subnet prefix differs 0x%"PRIx64" vs 0x%"PRIx64" and FLID routing is disabled"